numba/targets/codegen.py

```diff
@@ -789,7 +789,15 @@ def _customize_tm_options(self, options):
         # As long as we don't want to ship the code to another machine,
         # we can specialize for this CPU.
         options['cpu'] = self._get_host_cpu_name()
-        options['reloc'] = 'default'
+        # LLVM 7 change: # https://reviews.llvm.org/D47211#inline-425406
+        # JIT needs static relocation on x86*
+        # native target is already initialized from base class __init__
+        arch = ll.Target.from_default_triple().name
+        if arch.startswith('x86'): # one of x86 or x86_64
+            reloc_model = 'static'
+        else:
+            reloc_model = 'default'
+        options['reloc'] = reloc_model
```

Comment:
Note: symptom of reloc-model going wrong can be segfault due to incorrect relocation because GOT base is 0 and `symbol@GOT` is translated to just the GOT-offset.